lib/model.js

```diff
@@ -4052,29 +4052,18 @@ class Model {
       const where = this.where(true);
 
       if (this.constructor._timestampAttributes.deletedAt && options.force === false) {
-        const attribute = this.constructor.rawAttributes[this.constructor._timestampAttributes.deletedAt];
-        const field = attribute.field || this.constructor._timestampAttributes.deletedAt;
-        const values = Utils.mapValueFieldNames(this.dataValues, this.changed() || [], this.constructor);
-
-        values[field] = new Date();
-        where[field] = Object.prototype.hasOwnProperty.call(attribute, 'defaultValue')
+        const attributeName = this.constructor._timestampAttributes.deletedAt;
+        const attribute = this.constructor.rawAttributes[attributeName];
+        const defaultValue = Object.prototype.hasOwnProperty.call(attribute, 'defaultValue')
           ? attribute.defaultValue
           : null;
+        const currentValue = this.getDataValue(attributeName);
+        if (currentValue == defaultValue || _.isEqual(currentValue, defaultValue)) {
```

Comment:
Needed to add `==` check for non-Postgres DBs. For some reason they can return `undefined` for `currentValue`, which is not `_.isEqual` to `null`.

Reply:
Linting didn't like the extra parens around `currentValue == null && defaultValue == null` so I had to move it out into a variable:
```js
const undefinedOrNull = currentValue == null && defaultValue == null;
if (undefinedOrNull || _.isEqual(currentValue, defaultValue))
```